ports/starlink-ast/portfile.cmake

```
@@ -0,0 +1,56 @@
+vcpkg_fail_port_install(ON_TARGET "Linux" "OSX")
+
+vcpkg_download_distfile(ARCHIVE
+    URLS "https://github.com/Starlink/ast/releases/download/v9.2.4/ast-9.2.4.tar.gz"
+    FILENAME "ast-9.2.4.tar.gz"
+    SHA512 84e6f243e6d9d77328b73b97355feb3990307fb9c8f9b2f30344d71e2f5e63a849cdce0090ff5b7cc16028e12d68516c885b13d76db841072c9d1d06a7742a9e
+)
+
+vcpkg_extract_source_archive_ex(
+    OUT_SOURCE_PATH SOURCE_PATH
+    ARCHIVE ${ARCHIVE}    
+)
+
+set(CONFIGURE_OPTIONS "--disable-shared --without-fortran star_cv_cnf_trail_type=long star_cv_cnf_f2c_compatible=no")
+
+if ("yaml" IN_LIST FEATURES)
+    set(CONFIGURE_OPTIONS "${CONFIGURE_OPTIONS} --with-yaml")
+else()
+    set(CONFIGURE_OPTIONS "${CONFIGURE_OPTIONS} --without-yaml")
+endif()
+
+if ("pthreads" IN_LIST FEATURES)
+    set(CONFIGURE_OPTIONS "${CONFIGURE_OPTIONS} --with-pthreads")
+else()
+    set(CONFIGURE_OPTIONS "${CONFIGURE_OPTIONS} --without-pthreads")
+endif()
+
+
+set(CONFIGURE_OPTIONS_DEBUG " CFLAGS='-DCMINPACK_NO_DLL -O0 -g ${VCPKG_C_FLAGS} ${VCPKG_C_FLAGS_DEBUG}'")
+set(CONFIGURE_OPTIONS_RELEASE " CFLAGS='-DCMINPACK_NO_DLL -O2 ${VCPKG_C_FLAGS} ${VCPKG_C_FLAGS_RELEASE}'")
+
+set(RELEASE_TRIPLET ${TARGET_TRIPLET}-rel)
+set(DEBUG_TRIPLET ${TARGET_TRIPLET}-dbg)
+
+if(VCPKG_TARGET_ARCHITECTURE STREQUAL "x64")
+    set(CONFIGURE_OPTIONS "${CONFIGURE_OPTIONS} --host=x86_64-w64-mingw32")
+else()
+    set(CONFIGURE_OPTIONS "${CONFIGURE_OPTIONS} --host=i686-w64-mingw32")
+endif()
+
+vcpkg_acquire_msys(MSYS_ROOT PACKAGES make automake1.16 perl)    
+vcpkg_configure_make(
+    SOURCE_PATH "${SOURCE_PATH}"
+    USE_WRAPPERS
+    OPTIONS ${CONFIGURE_OPTIONS}
+    OPTIONS_RELEASE ${CONFIGURE_OPTIONS_RELEASE}
+    OPTIONS_DEBUG ${CONFIGURE_OPTIONS_DEBUG}
+)

```

```suggestion
vcpkg_acquire_msys(MSYS_ROOT PACKAGES make automake1.16 perl)    
vcpkg_configure_make(
    SOURCE_PATH "${SOURCE_PATH}"
    USE_WRAPPERS
    DETERMINE_BUILD_TRIPLET
    OPTIONS ${CONFIGURE_OPTIONS}
    OPTIONS_RELEASE ${CONFIGURE_OPTIONS_RELEASE}
    OPTIONS_DEBUG ${CONFIGURE_OPTIONS_DEBUG}
)
```

Can you try is this also works (adding `DETERMINE_BUILD_TRIPLET` to `vcpkg_configure_make` instead of set `--host` manually). Then adding the linux/osx support is easier in the future